resources/recipes/prepare_system.rb

```diff
@@ -130,8 +130,7 @@
 #getting total system memory less 10% reserved by system
 sysmem_total = (node["memory"]["total"].to_i * 0.90).to_i
 #node attributes related with memory are changed inside the function to have simplicity using recursivity
-memory_services(sysmem_total)
-
+memory_services(sysmem_total, ['chef-client']) # Don't assign memory to chef because the service will get handled
```

Comment:
since `memory_services` was already prepared for this cases, we sould  maybe create a new attr with the excluded services, like node["excluded_mem_services"]`

Reply:
we can control in the future what services are excluded with that, access from other nodes, and also for being able to write to that attr from external sources

Reply:
for example `resources/attributes/default.rb`

`default["redborder"]["excluded_memory_services"] = %w[chef-client]`